src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -338,10 +339,35 @@ def _alloc_from_content(self, user_config_file=None, config=None):
             jobs.append(job)
 
         self.forward_model = ForwardModel(jobs=jobs)
+
+        if (
+            ConfigKeys.JOBNAME in config_content_dict
+            and ConfigKeys.ECLBASE in config_content_dict
+        ):
+            logger.warning(
+                "Can not have both JOBNAME and ECLBASE keywords. "
+                "ECLBASE ignored, using JOBNAME with value "
+                f"`{config_content_dict[ConfigKeys.JOBNAME]}` instead"
+            )
+
         self.model_config = ModelConfig(
-            data_root=self.config_path,
+            config_dir=self.config_path,
+            num_realizations=config_content_dict.get(ConfigKeys.NUM_REALIZATIONS),
             refcase=self.ensemble_config.refcase,
-            config_content=user_config_content,
+            data_root=config_content_dict.get(ConfigKeys.DATAROOT),
+            ens_path=config_content_dict.get(ConfigKeys.ENSPATH),
+            history_source=HistorySourceEnum.from_string(
+                config_content_dict.get(ConfigKeys.HISTORY_SOURCE)
+            )
+            if ConfigKeys.HISTORY_SOURCE in config_content_dict
+            else None,
+            runpath_format_string=config_content_dict.get(ConfigKeys.RUNPATH),
+            jobname_format_string=config_content_dict.get(
+                ConfigKeys.JOBNAME, config_content_dict.get(ConfigKeys.ECLBASE, None)
+            ),
+            gen_kw_export_name=config_content_dict.get(ConfigKeys.GEN_KW_EXPORT_NAME),
+            obs_config_file=config_content_dict.get(ConfigKeys.OBS_CONFIG),
+            time_map_file=config_content_dict.get(ConfigKeys.TIME_MAP),
```

Comment:
I think if you add a class method in `ModelConfig` that does this type of initialization this will clean a bit `ResConfig`

it might look something like this:

```python
@classmethod
def from_dict(config_dict:dict) -> ModelConfig:
    return ModelConfig(
            config_dir=self.config_path,
            num_realizations=config_content_dict.get(ConfigKeys.NUM_REALIZATIONS),
            refcase=self.ensemble_config.refcase,
            data_root=config_content_dict.get(ConfigKeys.DATAROOT),
            ens_path=config_content_dict.get(ConfigKeys.ENSPATH),
            history_source=HistorySourceEnum.from_string(
                config_content_dict.get(ConfigKeys.HISTORY_SOURCE)
            )
            if ConfigKeys.HISTORY_SOURCE in config_content_dict
            else None,
            runpath_format_string=config_content_dict.get(ConfigKeys.RUNPATH),
            jobname_format_string=config_content_dict.get(
                ConfigKeys.JOBNAME, config_content_dict.get(ConfigKeys.ECLBASE, None)
            ),
            gen_kw_export_name=config_content_dict.get(ConfigKeys.GEN_KW_EXPORT_NAME),
            obs_config_file=config_content_dict.get(ConfigKeys.OBS_CONFIG),
            time_map_file=config_content_dict.get(ConfigKeys.TIME_MAP),
    
```

Reply:
sure thing! and the code becomes oh so much dryer, love it ^^ thank you!
it is done!